Framework/PythonInterface/plugins/algorithms/MaskShadowedPixels.py

```diff
@@ -0,0 +1,178 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+#pylint: disable=no-init,invalid-name
+import mantid
+import numpy as np
+
+
+class MaskShadowedPixels(mantid.api.PythonAlgorithm):
+    def category(self):
+        return "DataHandling\\Nexus"
+
+    def seeAlso(self):
+        return ["MaskShadowedPixels"]
+
+    def name(self):
+        return "MaskShadowedPixels"
+
+    def summary(self):
+        return "Generate maks file for NOMAD to exclude shadowed pixels."
+
+    def PyInit(self):
+        self.declareProperty(mantid.api.WorkspaceProperty('InputWorkspace', '', mantid.kernel.Direction.Input), "Workspace to save")
+        self.declareProperty(mantid.api.FileProperty('OutputFilename', '', action=mantid.api.FileAction.Save, extensions=['.gss']),
```

Comment:
I don't recall gss file as one of the desired output for this user story.  Is the requirement changed along the way?